### PR TITLE
Allow apps to log actions into the audit_log

### PR DESCRIPTION
**Pull request by nickvergessen**
Signed-off-by: Joas Schilling <coding@schilljs.com>

### CODE REVIEW

**Review by rullzer — APPROVED** [review]

File: lib/public/Log/AuditEvent.php [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCP\Log;
```
Comment (ChristophWurst) [inline lib/public/Log/AuditEvent.php:27]: how about a more specific namespace? OCP\Log\Audit for example?

File: lib/public/Log/AuditEvent.php [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+use OCP\EventDispatcher\Event;
+
+/**
+ * Emitted when the admin_audit app should log an entry
+ *
+ * @since 22.0.0
+ */
+class AuditEvent extends Event {
```
Comment (ChristophWurst) [inline lib/public/Log/AuditEvent.php:36]: it's a tradition for me to point out this doesn't comply with our naming convention

how about CriticalActionPerformedEvent or something that tells you *what happened*?

**Comment by ChristophWurst** [comment]
I like this change though. This will help clean up some old hooks from the current admin audit logging and it will open the door for more apps to integrate with this :+1:

File: lib/public/Log/AuditEvent.php (hunk 2 above)
Comment (nickvergessen) [reply lib/public/Log/AuditEvent.php]: well the "what" doesnt work as its anything apps want to log. So `CriticalActionPerformedEvent` it is

**Review by ChristophWurst — APPROVED** [review]

File: lib/public/Log/Audit/CriticalActionPerformedEvent.php [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+	private $parameters;
+
+	/** @var bool */
+	private $obfuscateParameters;
+
+	/**
+	 * @param string $logMessage
+	 * @param array $parameters
+	 * @param bool $obfuscateParameters
+	 * @since 18.0.0
```
Comment (blizzz) [inline lib/public/Log/Audit/CriticalActionPerformedEvent.php:51]: 18 o.O

Comment (blizzz) [inline lib/public/Log/Audit/CriticalActionPerformedEvent.php:51]:
```suggestion
	 * @since 22.0.0
```

**Review by blizzz — APPROVED** [review]

**Comment by nickvergessen** [comment]
Documentation in https://github.com/nextcloud/documentation/pull/9310

**Pull request merged**